MaxText/inference_utils.py

```diff
@@ -143,6 +143,8 @@ def sampling(logits, rng, algorithm, topk=0, nucleus_topp=0, temperature=1.0):
     return sample_nucleus_topp_logits(logits, nucleus_topp, temperature, rng)
   elif algorithm == "topk":
     return sample_topk_logits(logits, topk, temperature, rng)
+  elif algorithm == "stochastic":
+    return sample_topk_topp_weighted(logits, topk, nucleus_topp, temperature, rng)
```

Comment:
I would make either the algorithm name or the function name more similar to the other one. I don't think it's wrong this way, but it's not immediately obvious to me that "stochastic" and "sample_topk_topp_weighted" are associated with each other. Or a little comment explaining how they are related is an option too. 

If you are feeling extra brave a little snippet comment for each of the algos could be very helpful. But that's not necessarily your responsibility.

Reply:
Yes, I have the same feeling that "stochastic" that is not representative the combined algorithm. Maybe "dynamic" or "multi-mode"?

Also, missing greedy?